### PR TITLE
fix: skip delete legacy webhook if we have no permissions

### PR DESCRIPTION
**Pull request by TheSpiritXIII**

### CODE REVIEW

File: e2e/webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	if err := clientSet.RbacV1().ClusterRoles().Delete(ctx, "gmp-system:operator:webhook-admin", metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error deleting cluster role: %s", err)
+	}
+	if err := clientSet.RbacV1().ClusterRoleBindings().Delete(ctx, "gmp-system:operator:webhook-admin", metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error deleting cluster role binding: %s", err)
+	}
+
+	// Restart the GMP operator could already be healthy before we deleted the RBAC policies.
```
Comment (macxamin) [inline e2e/webhook_test.go:47]: Can you rephrase this comment? I'm having trouble understanding it.

File: e2e/webhook_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		}
+
+		t.Logf("waiting for operator logs to contain RBAC message")
+		return strings.Contains(logs, "delete legacy ValidatingWebHookConfiguration was not allowed"), nil
+	}); err != nil {
+		t.Fatalf("unable to check operator logs: %s", err)
+	}
+}
+
+func deploymentRestart(ctx context.Context, kubeClient kubernetes.Interface, name, namespace string) error {
```
Comment (macxamin) [inline e2e/webhook_test.go:95]: IIUC when the pod is deleted, it gets recreated automatically? can you please add a comment here.

File: e2e/webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	for i := range pods.Items {
+		pod := pods.Items[i]
+		if err := kubeClient.CoreV1().Pods(namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func deploymentPods(ctx context.Context, kubeClient kubernetes.Interface, name, namespace string) (*v1.PodList, error) {
```
Comment (macxamin) [inline e2e/webhook_test.go:110]: How is this function different from kube.deploymentPods?

**Review by macxamin — APPROVED** [review]
LGTM! Thank you sou much for the fast fix!

File: e2e/webhook_test.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+// TestWebhooksNoRBAC validates that the operator works without any webhook RBAC policies.
+func TestWebhooksNoRBAC(t *testing.T) {
+	ctx := context.Background()
+	clientSet, _, err := setupCluster(ctx, t)
+	if err != nil {
+		t.Fatalf("error instantiating clients. err: %s", err)
+	}
+
+	if err := clientSet.RbacV1().ClusterRoles().Delete(ctx, "gmp-system:operator:webhook-admin", metav1.DeleteOptions{}); err != nil {
```
Comment (bwplotka) [inline e2e/webhook_test.go:39]: Nice
